pyinfraformat/core/coord_utils.py

```diff
@@ -0,0 +1,323 @@
+"""Utilites to handle coordinates and transformations."""
+import logging
+import re
+from copy import deepcopy
+from pyproj import Transformer
+import numpy as np
+from .core import Holes, Hole
+
+__all__ = ["project_holes"]
+
+logger = logging.getLogger("pyinfraformat")
+
+TRANSFORMERS = {}  # dict of pyproj Transformers, key (input, output)
+
+
+def coord_string_fix(input_string):
+    """Try to fix coordinate systems string into machine readable."""
+    common_separators = r"[,. :\_-]"
+    if len(input_string) <= 4:
+        input_string = "ETRS-" + input_string
+    return "-".join(re.split(common_separators, input_string.upper()))
+
+
+def change_x_to_y(holes):
+    """Change holes x to y and y to x. Creates a deepcopy."""
+    holes_copy = deepcopy(holes)
+    for hole in holes_copy:
+        if (
+            hasattr(hole, "header")
+            and hasattr(hole.header, "XY")
+            and "X" in hole.header.XY
+            and "Y" in hole.header.XY
+        ):
+            hole.header.XY["X"], hole.header.XY["Y"] = (hole.header.XY["Y"], hole.header.XY["X"])
+    return holes_copy
+
+
+def proj_espoo(x, y):
+    """Project Espoo vvj coordinates into ETRS-GK24 (EPSG:3878).
+
+    https://www.espoo.fi/fi-FI/Asuminen_ja_ymparisto/Rakentaminen
+    /Kiinteiston_muodostus/Koordinaattiuudistus(19923)
+    """
+    # pylint: disable=invalid-name
+    output_epsg = "EPSG:3878"
+    a = 6599858.007479810200000
+    b = 24499824.978235636000000
+    c = 0.999998786628487
+    d = 0.000020762261526
+    e = -0.000014784506306
+    f = 0.999996546603269
+    x, y = a + c * x + d * y, b + e * x + f * y
+    return x, y, output_epsg
+
+
+def proj_helsinki(x, y):
```

Comment:
Do we have possibility for 2 / 3 different transformations?

What about `z`?

Reply:
Yea, good idea. Needs a bit of coding.

https://www.maanmittauslaitos.fi/kartat-ja-paikkatieto/asiantuntevalle-kayttajalle/koordinaattimuunnokset

Reply:
Done.